src/configure/clients/devOps/azureDevOpsClient.ts

```diff
@@ -135,8 +207,77 @@ export class AzureDevOpsClient {
         }
     }
 
+    public async validateOrganizationName(organizationName: string): Promise<string> {
+        let deferred = Q.defer<string>();
+        let accountNameRegex = new RegExp(/^[a-zA-Z0-9][a-zA-Z0-9-]*[a-zA-Z0-9]$|^[a-zA-Z]$/);
+
+        if(!organizationName || /^\\s/.test(organizationName) || /\\s$/.test(organizationName) || organizationName.indexOf("-") === 0 || !accountNameRegex.test(organizationName)) {
+            deferred.resolve(Messages.organizationNameStaticValidationMessage);
+        }
+
+        if(ReservedHostNames.indexOf(organizationName) >= 0) {
+            deferred.resolve(util.format(Messages.organizationNameReservedMessage, organizationName));
+        }
+        
+        let url = `https://app.vsaex.visualstudio.com/_apis/HostAcquisition/NameAvailability/${organizationName}`;
+
+        this.restClient.sendRequest<any>(<UrlBasedRequestPrepareOptions>{
+            url: url,
+            headers: {
+                "Content-Type": "application/json",
+                "Accept": "api-version=5.0-preview.1"
+            },
```

Comment:
I think the miss here is that these conditions should be under else block. because if static validation fails, then we are immediately resolving the promise with validation message. We don't want the REST call to go if static validation fails.